src/main/java/io/github/vcuswimlab/stackintheflow/controller/component/stat/tags/UserTagStatComponent.java

```diff
@@ -18,16 +23,48 @@
 
     private TagStatComponent tagStatComponent;
     private PersonalSearchModel searchModel;
+	private L2HPredictor l2hPredictor;
     private Map<String, Integer> userStatMap;
 
     public UserTagStatComponent(TagStatComponent tagStatComponent, Project project) {
         this.tagStatComponent = tagStatComponent;
         this.userStatMap = ServiceManager.getService(project, PersistProfileComponent.class).getUserStatMap();
     }
 
+	public boolean createInitialTagPredictions(String filePath) {
+		try {
+			l2hPredictor.computeL2HPredictions(filePath, "./temp_predictions_file.txt", "./temp_text_data_file.txt");
```

Comment:
Yes, this is the design aspect I was most unhappy about. Segan only allows predictions to move in or out of local files and provides no methods or parameters to put them anywhere else. Ideally, they would go in a collection, but there's no easy way to do that. In theory, it's possible to allow this by duplicating a number of segan's classes and adding custom code paths that work with collections instead of files. However, it's unclear how difficult that would be given how segan's design is based entirely around local file I/O. I could look into it if you feel the benefits would outweigh the time spent, code bloat, and additional code/design complexity.

Reply:
@cooglejj can you do some research into using pipes (or FIFOs) instead of actual files? this way you won't have to manage creating and deleting them, plus it should have better performance. my quick google search indicates that these aren't available for Java, but there could be a library or something we can use. 

https://www.gnu.org/software/libc/manual/html_node/Pipes-and-FIFOs.html

Reply:
No problem. I looked into what segan did and created a custom version of their code that does everything in memory, which of course means no more temp files. It does add some extra code, but not as much as I originally thought.